File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantInvokeDynamicInfo.java
package com.github.zxh.classpy.classfile.constant;

import com.github.zxh.classpy.classfile.datatype.U2;

/*
CONSTANT_InvokeDynamic_info {
    u1 tag;
    u2 bootstrap_method_attr_index;
    u2 name_and_type_index;
}
*/
public class ConstantInvokeDynamicInfo extends ConstantInfo {

    {
        u2("bootstrap_method_attr_index");
        u2("name_and_type_index");
    }
    
    @Override
    protected String loadDesc(ConstantPool cp) {
        int nameAndTypeIndex = super.getUInt("name_and_type_index");
        return cp.getNameAndTypeInfo(nameAndTypeIndex).loadDesc(cp);
    }
    
}

File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantInvokeDynamicInfo.kt
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_InvokeDynamic_info {
    u1 tag;
    u2 bootstrap_method_attr_index;
    u2 name_and_type_index;
}
*/
class ConstantInvokeDynamicInfo : ConstantInfo() {

    init {
        u2("bootstrap_method_attr_index");
        u2("name_and_type_index");
    }
    
    override fun loadDesc(cp: ConstantPool): String {
        val nameAndTypeIndex = super.getInt("name_and_type_index");
        return cp.getNameAndTypeInfo(nameAndTypeIndex).loadDesc(cp);
    }
    
}
